design-documents/catalog-domain/product-data-object.md

```diff
@@ -0,0 +1,377 @@
+* `fixedTaxes` - fixed product taxes. Example WEEE tax.
+```json
+{
+  "type": "FixedTax",
+  "fields": [
+    {"name": "title", "type": "string"},
+    {"name": "amount", "type": "float"}
+  ]
+}
+```
+
+## Inventory attributes
+
+Catalog knows the bare minimum information about inventory,
+only data that we need to render product.
+
+```json
+{
+  "type": "Product",
+  "fields": [
+    {"name": "inStock", "type": "bool"},
+    {"name": "lowStock", "type": "bool"}
+  ]
+}
+```
+* `isStock` - is product in stock.
+* `lowStock` - is product in low stock.
+
+
+## Product Option & Values
+Product options explain how product can be configured before adding a product to cart.
+
+```json
+{
+  "type": "ProductOption",
+  "fields": [
+    {"name": "id", "type": "uuid"},
+    {"name": "isRequired", "type": "bool"},
+    {"name": "isMulti", "type": "bool"},
+    {"name": "name", "type": "string"},
+    {"name": "values", "type": "ProductOptionValue", "repeated": "true"}
+  ]
+}
+```
+* `id` - option identifier.
+* `isRequired` - is this options required.
+* `isMulti` - is the option allows multiple choices.
+* `name` - option name.
+* `values` - set of option values.
+
+```json
+{
+  "type": "ProductOptionValue",
+  "fields": [
+    {"name": "id", "type": "uuid"},
+    {"name": "value", "type": "string"},
+    {"name": "minimalPrice", "type": "Price"}
+  ]
+}
+```
+* `id` - value id.
+* `value` - option value.
+* `minimalPrice` - option real price. 
+
+## Product Images 
+
+```json
+{
+  "type": "Product",
+  "fields": [
+    {"name": "image", "type": "Image"},
+    {"name": "smallImage", "type": "Image"},
+    {"name": "thumbnailImage", "type": "Image"},
+    {"name": "swatchImage", "type": "Image"},
+    {"name": "mediaGallery", "type": "Image", "repeated": "true"}
+  ]
+}
+```
+* `image` - base product image.
+* `smallImage` - small product image.
+* `thumbnailImage` - thumbnail image.
+* `swatchImage` - image to represent a product with swatch functionality.
+* `mediaGallery` - set of images to be shown at the *PDP*.
+
+```json
+{
+  "type": "Image",
+  "fields": [
+    {"name": "url", "type": "string"},
+    {"name": "label", "type": "string"}
+  ]
+}
+```
+
+## [TBD]Product Videos
+```json
+{
+  "type": "Product",
+  "fields": [
+    {"name": "videos", "type": "Video", "repeated": "true"}
+  ]
+}
+```
+* `videos` - set of product videos.
+```json
+{
+  "type": "Video",
+  "fields": [
+    {"name": "url", "type": "string"},
+    {"name": "label", "type": "string"}
+  ]
+}
+```
+
+## Product Variations
+
+* *Product variation* is a representation of one or many product options that represent a dedicated simple product.
+
+```json
+{
+  "type": "Product",
+  "fields": [
+    {"name": "variations", "type": "ProductVariation", "repeated": "true"}
+  ]
+}
+```
+```json
+{
+  "type": "ProductVariation",
+  "fields": [
+    {"name": "id", "type": "string"},
+    {"name": "productId", "type": "string"},
```

Comment:
"productId" - is this a reference to real simple product? Then what is "selections"?